dht/i_partitioner.cc

```diff
@@ -300,19 +300,22 @@ std::strong_ordering ring_position_comparator_for_sstables::operator()(sstables:
 
 dht::partition_range
 to_partition_range(dht::token_range r) {
+    using bound = dht::partition_range::bound;
     using bound_opt = std::optional<dht::partition_range::bound>;
     auto start = r.start()
-                 ? bound_opt(dht::ring_position(r.start()->value(),
+                 ? bound_opt(bound(dht::ring_position(r.start()->value(),
                                                 r.start()->is_inclusive()
                                                 ? dht::ring_position::token_bound::start
-                                                : dht::ring_position::token_bound::end))
+                                                : dht::ring_position::token_bound::end),
+                         r.start()->is_inclusive()))
```

Comment:
We already have a mention of is_inclusive, isn't it sufficient? If it's exclusive, the ring position will start at the end bound.

Reply:
No, it is not enough, because if we disregard inclusiveness at the bound level, we can produce intersections like:
```
[{t1, end}, {t1, end}]
```

Reply:
Ouch, but please add tests.

That range is empty btw.

Reply:
My existing tests are catching this. In particular, the test which passes non-overlapping (but sharing bounds) range to the splitter, is producing such empty ranges without the fix. That test is very sensitive to correctness around touching bounds.